pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -639,6 +639,14 @@ type PoolSpec struct {
 	Quotas QuotaSpec `json:"quotas,omitempty"`
 }
 
+// NamedPoolSpec represents the named ceph pool spec
+type NamedPoolSpec struct {
+	// Name of the pool
+	Name string `json:"name,omitempty"`
```

Comment:
I'm still struggling a bit with calling this `name` since the name ends up being partly generated. At the end of the day, I'm not sure something like `generateName` is very useful here and that improving the docs is the best thing to do.

We still might want to make this struct CephFilesystem specific (i.e., not reusing the named pool spec for CephBlockPool) and having the doc text be something like "unique identifier for the pool within the filesystem. will be prefixed with the metadata name." We should also mention that we do most definitely recommend setting the pool name here since it helps resolve errors that could arise from altering the pool list.

Reply:
I like the NamedPoolSpec to use both for the CephBlockPool and the CephFilesystem. In the Filesystem docs sounds good to just be clear that the filesystem name is prepended to the name.

Reply:
> In the Filesystem docs sounds good to just be clear that the filesystem name is prepended to the name.

Yeah, we may add note that name for pool will be prepended to the filesystem name.